OpenRA.Mods.Common/Traits/Player/ModularBot.cs

```diff
@@ -25,15 +25,16 @@ public sealed class ModularBotInfo : TraitInfo, IBotInfo
 		[Desc("Internal id for this bot.")]
 		public readonly string Type = null;
 
+		[TranslationReference]
 		[Desc("Human-readable name this bot uses.")]
-		public readonly string Name = "Unnamed Bot";
+		public readonly string Name = null;
 
 		[Desc("Minimum portion of pending orders to issue each tick (e.g. 5 issues at least 1/5th of all pending orders). Excess orders remain queued for subsequent ticks.")]
 		public readonly int MinOrderQuotientPerTick = 5;
 
 		string IBotInfo.Type => Type;
 
-		string IBotInfo.Name => Name;
+		string IBotInfo.Name => TranslationProvider.GetString(Name);
```

Comment:
translating here will very likely cause issues as this name is used not only for UI

Reply:
Not sure I see the issue here. This maintains the previous behaviour where this field returned the display name.

And this doesn't seem like a performance issue, none of the callers looks like they'd get called frequently.

Reply:
it's more that there can be issues with syncing with other players as they might be using a different language

Reply:
Are you sure that you aren't getting this mixed up with `IBotInfo.Type`? `Name` is stupposed to only be used for UI, and `Type` for matching to internal things.

Reply:
I think Pun is right:

`ResolvePlayerName` uses the name for checking if bots are of the same type. That sounds like it would be safer if it used the type.

`SanitizedPlayerName` stops people using a bot name, if the bot names can differ in language that might need changing somehow.

Reply:
`SanitizedPlayerName` is an interesting case, we probably want some new plumbing to query a dictionary of language id -> string for a given translation id.

Reply:
Dropping the bot name check seems reasonable.

If we did want to have some sort of safety here, adding a bot icon or some other UI affordance would allow distinuisghing between human and AI players. Such an affordance would also be helpful than the current don't-reuse-name check, e.g. if a human player uses a visually similar yet slightly different name to one of the bots.

Reply:
That makes sense - we already do this for registered vs anonymous players, so that indicator could grow a new 🤖 state.

Reply:
but it still seems wrong to have the names of the bots be determined by the language the admin runs. It could have font issues as well

Reply:
Indeed, bot names should be resolved client-side. This should be simple enough, since the bot type is specified on the client info.